pkg/storegateway/bucket.go

```diff
@@ -709,9 +709,9 @@ func mapSeriesError(err error) error {
 	case errors.As(err, &stGwErr):
 		switch cause := stGwErr.errorCause(); cause {
 		case mimirpb.INSTANCE_LIMIT:
-			return globalerror.NewErrorWithGRPCStatus(stGwErr, codes.Unavailable, &mimirpb.ErrorDetails{Cause: cause})
+			return globalerror.WrapErrorWithGRPCStatus(stGwErr, codes.Unavailable, &mimirpb.ErrorDetails{Cause: cause}).Err()
```

Comment:
`.Err()` is not mandatory, it is just used to remove semantic connection with `stGwErr`